src/api.rs

```diff
@@ -460,12 +455,7 @@ impl Api {
 
     /// Runs a query against the MediaWiki API, using `method` GET or POST.
     /// Parameters are a hashmap; `format=json` is enforced.
-    async fn query_api_json_mut(
-        &mut self,
-        params: &Params,
-        method: Method,
-    ) -> Result<Value, Error> {
-        let mut params = params.clone();
+    async fn query_api_json_mut(&mut self, mut params: Params, method: Method) -> Result<Value> {
```

Comment:
I like how you've shifted the burden of `.clone()`-ing onto the caller here.

Reply:
Thanks! The downside is that the signatures are no longer consistent (some accept `&Params` and some accept `Params`).